modules/4337/test/erc4337/ReferenceEntryPoint.spec.ts

```diff
@@ -215,6 +215,208 @@ describe('Safe4337Module - Reference EntryPoint', () => {
+        initCode: daughterSafe.getInitCode(),
+      },
+    )
+
+    const opData = calculateSafeOperationData(await validator.getAddress(), safeOp, await chainId())
+    const signature = buildSignatureBytes([
+      {
+        signer: parentSafe.address,
+        data: await user.signTypedData(
+          {
+            verifyingContract: parentSafe.address,
+            chainId: await chainId(),
+          },
+          {
+            SafeMessage: [{ type: 'bytes', name: 'message' }],
+          },
+          {
+            message: opData,
+          },
+        ),
+        dynamic: true,
+      },
+    ])
+    const userOp = buildPackedUserOperationFromSafeUserOperation({
+      safeOp,
+      // Replace the 2nd word of static part of signature containing the pointer to dynamic part with invalid pointer value
+      signature: signature.slice(0, 67) + '00'.padStart(64, '0') + signature.slice(131),
+    })
+
+    await expect(entryPoint.handleOps([userOp], await relayer.getAddress()))
+      .to.be.revertedWithCustomError(entryPoint, 'FailedOp')
+      .withArgs(0, 'AA24 signature error')
+  })
+
+  it('should revert when expected offset and given dynamic pointer in signatures are not equal - Smart contract signature (NOTE: would require a staked paymaster for ERC-4337)', async () => {
+    const { user, relayer, safe: parentSafe, validator, entryPoint, safeGlobalConfig } = await setupTests()
+
+    await parentSafe.deploy(user)
+    const daughterSafe = Safe4337.withSigner(parentSafe.address, safeGlobalConfig)
+
+    const accountBalance = ethers.parseEther('1.0')
+    await user.sendTransaction({ to: daughterSafe.address, value: accountBalance })
+    expect(await ethers.provider.getBalance(daughterSafe.address)).to.be.eq(accountBalance)
+
+    const transfer = ethers.parseEther('0.1')
+    const safeOp = buildSafeUserOpTransaction(
+      daughterSafe.address,
+      user.address,
+      transfer,
+      '0x',
+      '0x0',
+      await entryPoint.getAddress(),
+      false,
+      false,
+      {
+        initCode: daughterSafe.getInitCode(),
+      },
+    )
+
+    const opData = calculateSafeOperationData(await validator.getAddress(), safeOp, await chainId())
+    const signature = buildSignatureBytes([
+      {
+        signer: parentSafe.address,
+        data: await user.signTypedData(
+          {
+            verifyingContract: parentSafe.address,
+            chainId: await chainId(),
+          },
+          {
+            SafeMessage: [{ type: 'bytes', name: 'message' }],
+          },
+          {
+            message: opData,
+          },
+        ),
+        dynamic: true,
+      },
+    ])
+
+    const userOp = buildPackedUserOperationFromSafeUserOperation({
+      safeOp,
+      // Replace the 2nd word of static part of signature containing the pointer to dynamic part with invalid pointer value
+      signature:
+        signature.slice(0, 66) +
+        ((signature.length - 2) / 2).toString(16).padStart(64, '0') +
```

Comment:
I think this offset is wrong - it points to the start of the padding and not the length. Notably this test *also* fails with `main` branch contracts, and is not covering the case of having a valid signature with extra bytes in between the dynamic signatures.

Reply:
In the current test suite, if we apply this diff:

```diff
diff --git a/modules/4337/contracts/Safe4337Module.sol b/modules/4337/contracts/Safe4337Module.sol
index 485ca91..62cea2a 100644
--- a/modules/4337/contracts/Safe4337Module.sol
+++ b/modules/4337/contracts/Safe4337Module.sol
@@ -246,7 +246,7 @@ contract Safe4337Module is IAccount, HandlerContext, CompatibilityFallbackHandle
             }
             /* solhint-enable no-inline-assembly */
             // If the signature data pointer is not pointing to the expected location, return false.
-            if (!pointsAtEnd) return false;
+            //if (!pointsAtEnd) return false;
         }
         isValid = signatures.length <= expectedOffset;
     }

```

The tests still pass - so we aren't covering this path fully.

Reply:
https://github.com/safe-global/safe-modules/pull/453/commits/42b216e6f756cb43fe396f1947367fe0898a9584
The test passes because ` isValid = signatures.length <= expectedOffset;` check prevents it. I also added signature  length manipulation check which justifies existence of `if (!pointsAtEnd) return false;`.

Still commenting the statement will lead to test to pass because Safe reverts while checking signatures. If we want failing test on removing `if (!pointsAtEnd) return false;`  then we need to use a Mock contract that passes signature checks even on invalid input.